lib/navigation/navigator.cpp

```diff
@@ -20,31 +20,23 @@ void Navigator::navigate(){
   */
 }
 
-void Navigator::publishTrajectory(){
-  /*
-  TODO: implement
-
-  idk how ros works but make sure everyone who needs to know trajectory has it
-  */
-}
 
-void Navigator::setImuData(const core::ImuData imu_data){
-  imu_data_.at(0) = static_cast<nav_t>(imu_data.imu0);
-  imu_data_.at(1) = static_cast<nav_t>(imu_data.imu1);
-  imu_data_.at(2) = static_cast<nav_t>(imu_data.imu2);
-  imu_data_.at(3) = static_cast<nav_t>(imu_data.imu3);
+void Navigator::setImuData(const core::RawImuData imu_data){
+  for(size_t i = 0; i < core::kNumImus; ++i){
+    raw_imu_data_.at(i) = imu_data.at(i);
+  }
 }
 
-void Navigator::setEncoderData(const core::WheelEncoderData encoder_data){
-  encoder_data_.at(0) = encoder_data.encoder0;
-  encoder_data_.at(1) = encoder_data.encoder1;
-  encoder_data_.at(2) = encoder_data.encoder2;
-  encoder_data_.at(3) = encoder_data.encoder3;
+void Navigator::setEncoderData(const core::EncoderData encoder_data){
+  for(size_t i = 0; i < core::kNumEncoders; ++i){
```

Comment:
`std::copy(encoder_data.begin(), encoder_data.end(), raw_encoder_data.begin());`

Reply:
Actually, since they have the same type just `raw_encoder_data_ = encoder_data` is good enough; the above is for cases where you may be going from `std::array` to `std::vector` or similar.

You may have to drop the `const` here, which is fine; or add a reference.